wondrous-app/components/OnboardingDao/StepWrapper/component.tsx

```diff
@@ -0,0 +1,149 @@
+import { CircularProgress } from '@mui/material';
+import LeftArrowIcon from 'components/Icons/leftArrow';
+import { useFormikContext } from 'formik';
+import { mapKeys, some } from 'lodash';
+import { useRouter } from 'next/router';
+import { useEffect } from 'react';
+import { onboardingDaoValueLocalStorageKey } from '../constants';
+import {
+  BackButton,
+  ButtonWrapper,
+  CloseButton,
+  CloseButtonIcon,
+  ComponentWrapper,
+  ContinueButton,
+  FooterWrapper,
+  FormWrapper,
+  HeaderWrapper,
+  LaterButton,
+  StepIndicatorDone,
+  StepIndicatorEmpty,
+  StepIndicatorFilled,
+  StepIndicatorWrapper,
+  Subtitle,
+  Title,
+  Wrapper,
+  WrapperLoadingCircularProgress,
+} from './styles';
+
+const NO_OF_STEPS = 5;
+
+export const OnboardingStepIndicator = ({ step }) => {
+  const rangeIndicator = (start, end, Component) =>
+    Array.from({ length: start - end }, (_, i) => <Component key={i} />);
+  return (
+    <StepIndicatorWrapper>
+      {rangeIndicator(step, 1, StepIndicatorDone)}
+      <StepIndicatorFilled />
+      {rangeIndicator(NO_OF_STEPS, step, StepIndicatorEmpty)}
+    </StepIndicatorWrapper>
+  );
+};
+
+const BackButtonWrapper = ({ step, handleStep }) => {
+  if (step === 1) return <ButtonWrapper />;
+  return (
+    <ButtonWrapper>
+      <BackButton onClick={() => handleStep({ action: 'back' })}>
+        <LeftArrowIcon />
+      </BackButton>
+    </ButtonWrapper>
+  );
+};
+
+const LaterButtonWrapper = ({ step, handleStep, hideLater }) => {
+  if (step === NO_OF_STEPS || hideLater) return null;
+  return <LaterButton onClick={() => handleStep({ action: 'next' })}>Later</LaterButton>;
+};
+
+const useValidateStep = (fields) => {
+  const { errors, setFieldTouched, validateForm } = useFormikContext();
+  const touchFields = () => mapKeys(fields, (_, key) => setFieldTouched(key));
+  const hasError = some(fields, (_, key) => errors[key]);
+  useEffect(() => {
+    validateForm();
+  }, [validateForm]);
+  return { touchFields, errors, hasError };
+};
+
+const ContinueButtonWrapper = ({ step, hoverContinue, handleStep, fields = {} }) => {
+  const { touchFields, hasError } = useValidateStep(fields);
+  const handleOnClick = (e) => {
+    e.preventDefault();
+    touchFields();
+    handleStep({ action: 'next', hasError });
```

Comment:
can we have something like const ACTION_TYPES = {NEXT: 'next', BACK: 'back' } and use that instead of the string?